platform/osgi-platform/src/main/java/org/motechproject/server/osgi/status/PlatformStatusManagerImpl.java

```diff
@@ -87,7 +118,33 @@ private void handleContextFailedEvent(OsgiBundleContextFailedEvent event) {
         LOGGER.debug("Received context closed event {} from {}", event, symbolicName);
         LOGGER.info("{} failed to start due to {}", symbolicName, failureCauseMsg);
 
+        checkBlueprintBundle(event.getBundle());
+
         platformStatus.removeStartedBundle(symbolicName);
         platformStatus.addContextError(symbolicName, failureCauseMsg);
     }
+
+    private void checkOsgiBundle(Bundle bundle) {
+        if (osgiBundles.contains(bundle)) {
+            osgiBundles.remove(bundle);
+            checkStartedBundle();
+        }
+    }
+
+    private void checkBlueprintBundle(Bundle bundle) {
+        if(blueprintBundles.contains(bundle)) {
+            blueprintBundles.remove(bundle);
+            checkStartedBundle();
+        }
+    }
+
+    private void checkStartedBundle() {
```

Comment:
missleading method name, this doesn't check if a bundle is started, but checks if all bundles have started